fix: adjust number formatters + "Max" spending limit

iamacook opened the pull request:
## What it solves

Number formatting and "Max" spending limit.

## How this PR fixes it

The number formatters have been updated to use `ethers`' `formatUnits` instead of `toDecimals` and the "Max" token amount has been removed from the `<SpendingLimitForm>` field as it's possible to set higher amounts than that which is in the Safe.

## How to test it

1. Observe all numbers formatted correctly across the Safe, i.e. in the assets list, in high and low amount transactions and in modals. Choosing the "Max" amount when the current transaction has high or low amounts, formatted with `>` or `<` should not copy these symbols into their respective amount fields.
2. The "Max" button should no longer exist in the spending limits form.

iamacook commented on src/utils/formatters.ts:
```diff
@@ -2,12 +2,8 @@ import { BigNumberish, type BigNumber } from 'ethers'
 import { formatUnits, parseUnits } from 'ethers/lib/utils'
 import { formatAmount } from './formatNumber'
 
-export const toDecimals = (value: BigNumberish, decimals?: number | string): number => {
-  return Number(formatUnits(value, decimals))
-}
-
```
This was an unnnecessary wrapper as we only once need convert the formatted numbers to numbers.

katspaugh commented on src/components/common/SafeTokenWidget/index.tsx:
```diff
@@ -23,9 +23,8 @@ const SafeTokenWidget = () => {
-    toDecimals(safeBalance?.balance || '0', safeBalance?.tokenInfo.decimals),
-  ).toFixed(0)
+  const safeBalanceDecimals = Number(formatUnits(safeBalance?.balance || 0, safeBalance?.tokenInfo.decimals))
+  const flooeredSafeBalance = Math.floor(safeBalanceDecimals).toFixed(0)
```
Typo in floored

katspaugh commented:
A tests is failing.

What are the differences between formatDecimals and formatUnits?

iamacook commented:
> What are the differences between formatDecimals and formatUnits?

@usame-algan and I just went through every instance of number formatting and cleaned everything up a bit to be the following:

- `safeFormatUnits` for anything logic-related, returning `123`, `0.00000000001`, etc.
- `safeFormatAmount` for anything UI-related, returning `-< 0.00001`, `999T`, etc.
- `safeParseUnits` for converting to `BigNumbers`.

The following formatters were removed as the above now covers their logic:
- `toDecimals` = `safeFormatUnits`
- `formatDecimals` = `safeFormatAmount`
- `toWei` = `safeParseUnits`

katspaugh commented:
Awesome! To make it more obvious which function to use – safeFormatUnits vs safeFormatAmount – should we maybe rename `safeFormatAmount` to `visualFormatAmount` or `renderAmount` or some such?

iamacook commented:
> Awesome! To make it more obvious which function to use – safeFormatUnits vs safeFormatAmount – should we maybe rename `safeFormatAmount` to `visualFormatAmount` or `renderAmount` or some such?

Great suggestion! I've renamed it to `formatVisualAmount` and adjusted the explanatory comment above.

schmanu commented on src/utils/formatters.ts:
```diff
@@ -2,29 +2,27 @@ import { BigNumberish, type BigNumber } from 'ethers'
-
 const GWEI = 'gwei'
 
+// safeFormatUnits => "0.000000000001"
```
I would suggest to write a proper comment here which explains the difference between `formatVisualAmount` and `safeFormatUnits`.

i.e.
```ts
/**
 *  Formats a GWEI value to decimal with high precision. 
 *  The resulting string is not formatted with respect to the standard amount formatting. Use {@link formatVisualAmount} to format values with amount formatting.
 * @param value in GWEI
 * @param decimals decimals of the value in GWEI, 18 decimals are used if undefined
 * @returns formatted decimal value with high precision
 */
```

schmanu commented on src/utils/formatters.ts:
```diff
@@ -2,29 +2,27 @@ import { BigNumberish, type BigNumber } from 'ethers'
   }
 }
 
+// formatVisualAmount => -< 0.00001
```
Here I would prefer a similar comment.
These functions are easily mistaken.

schmanu reviewed (commented):
It's good that we streamlined the amount /value formatting! Nice work!
